pkg/store/storepb/custom.go

```diff
@@ -497,32 +497,45 @@ func (m *SeriesResponse) Marshal() (dAtA []byte, err error) {
 
 	var respBuf []byte
 
-	// No pool defined, allocate directly.
+	// Slow path with no sync.Pool.
 	if m.respPool == nil {
 		respBuf = make([]byte, size)
-	} else {
-		if m.respBuf == nil {
-			poolBuf := m.respPool.Get()
-			if poolBuf == nil {
-				respBuf = make([]byte, size)
-				m.respBuf = &respBuf
-			} else {
-				m.respBuf = poolBuf.(*[]byte)
-				respBuf = *m.respBuf
-			}
+		m.respBuf = nil
+
+		n, err := m.MarshalToSizedBuffer(respBuf)
+		if err != nil {
+			return nil, err
+		}
+		return respBuf[len(respBuf)-n:], nil
+	}
+
+	// Fast path with sync.Pool.
+	// m.respBuf must not be nil so that it would be returned to the pool.
```

Comment:
Does it need to non nil if we do the work below? Can we always do that and remove any outside call pool Get?

Reply:
I'm not sure I understood this comment. The callers using the old function pass `nil` thus we need to handle this case - it is not guaranteed that the pool exists

Reply:
I know, i thought we can simplify, and allow this method to `Get`, never pass it from outside - not a biggie